hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/compactions/Compactor.java

```diff
@@ -278,6 +280,21 @@ protected final StoreFileWriter createTmpWriter(FileDetails fd, boolean shouldDr
         HConstants.EMPTY_STRING);
   }
 
+  /**
+   * Default method for initializing a StoreFileWriter in the compaction process, this creates the
+   * resulting files on a temp directory. Therefore, upon compaction commit time, these files
+   * should be renamed into the actual store dir.
+   * @param fd the file details.
+   * @param shouldDropBehind boolean for the drop-behind output stream cache settings.
+   * @param major if compaction is major.
+   * @return Writer for a new StoreFile in the tmp dir.
+   * @throws IOException if it fails to initialise the writer.
+   */
+  protected StoreFileWriter initWriter(FileDetails fd, boolean shouldDropBehind, boolean major)
+    throws IOException {
+    return this.createTmpWriter(fd, shouldDropBehind, major);
```

Comment:
Why the 'this'? Seems unorthodox? We don't usually do this?

Reply:
Noted. Removing 'this' on next commit.